dakara_server/library/management/commands/createworks.py

```diff
@@ -0,0 +1,258 @@
+import os
+import sys
+import logging
+import importlib
+
+from django.core.management.base import BaseCommand, CommandError
+
+from library.models import (WorkType, WorkAlternativeTitle, Work)
+
+from .feed_components import default_work_parser
+
+# get logger
+logger = logging.getLogger(__name__)
+
+
+class WorkCreator:
+    """Work creator and updater. Create and update works in the
+    database provided a work file and a parser.
+
+    Args:
+        work_file (str): Absolute path of the file storing the works data.
+        parser (module): Custom python module used to extract data from file.
+        dry_run (bool): Run script in test mode.
+
+    About parser:
+        This module should define a method called `parse_work` which takes
+        a file path as argument and return a dictionnary with the following:
+            works (dict): keys are a query name worktype, values are
+            a list such that each element of this list is a dictionnary with
+            the following keys:
+                    title (str): title of the work
+                    subtitle (str): subtitle of the work
+                    alternative_titles (list): list of alternative names of
+                    a work
+    """
+    def __init__(
+            self,
+            work_file,
+            parser,
+            debug):
+        self.work_file = work_file
+        self.parser = parser
+        self.debug = debug
+
+    def remove_incorrect_works(self, work_listing):
+        """Remove works that have an incorrect structure in a list of works.
+
+        Return
+            A new list with the incorrect works removed."""
+
+        work_to_remove = []
+
+        logger.debug("Removing the incorrect work from the current work type.")
+        for dict_work in work_listing:
+
+            # check if it is a dictionary
+            if not isinstance(dict_work, dict):
+                work_to_remove.append(dict_work)
+                logger.debug("Incorrect work : value is not a dictionary.")
+                continue
+
+            # check if it has a title field
+            work_title = dict_work.get('title')
+            if not work_title:
+                work_to_remove.append(dict_work)
+                logger.debug("Incorrect work : no title field found")
+                continue
+
+            logger.debug("Work '{}' is correctly structured.".format(
+                    work_title))
+
+        return [work for work in work_listing if work not in work_to_remove]
+
+    def debug_parser_work(self, dict_work):
+        """Display debug messages on a specific work dictionarry.
+
+        Args
+            dict_work: dictionnary for which the check is done
+            (must have a 'title' field)
+        """
+        field_names = ('subtitle', 'alternative_titles')
+
+        # get the work title (should have been checked it exists
+        # before calling this method)
+        work_title = dict_work.get('title')
```

Comment:
The dictionary `get` method will always return something. Is the key does not exist, the default value is `None`. Since you have checked in `remove_incorrect_works` that the work has a non-falsy title, you can assume here that the title is present and non-empty…